airframe-fluentd/src/main/scala/wvlet/airframe/fluentd/FluencyClient.scala

```diff
@@ -22,12 +24,63 @@ case class FluencyConfig(
     // Use the extended EventTime timestamps
     // https://github.com/fluent/fluentd/wiki/Forward-Protocol-Specification-v1#eventtime-ext-format
     useExtendedEventTime: Boolean = false,
-    fluencyConfig: Fluency.Config = new Fluency.Config()
+    // Default value of following properties are all Fluency default configuration
+    maxBufferSize: Long = 512 * 1024 * 1024,
+    bufferChunkInitialSize: Int = 1024 * 1024,
+    bufferChunkRetentionSize: Int = 1024 * 1024 * 4,
+    bufferChunkRetentionTimeMillis: Int = 1000,
+    flushIntervalMillis: Int = 600,
+    fileBackupDir: String = null,
+    waitUntilBufferFlushed: Int = 60,
+    waitUntilFlusherTerminated: Int = 60,
+    jvmHeapBufferMode: Boolean = false,
+    errorHandler: ErrorHandler = null,
```

Comment:
I think we have no way to set this errorHandler value via Yaml-based configuration (e.g., using airframe-config) unless providing default implementations (e.g., defautErrorHandler, etc.) 

We have two approaches: passing Fluency configuration or passing Fluency itself in DI:
```
val fluency: Fluency = bind{ errorHandler: FluencyErrorHandler => Fluency.newClient.withErrorHandler(errorHandler).build } 
```

As I commented before https://github.com/wvlet/airframe/pull/368#issuecomment-457337067, using type safe builder interface is most flexible for such configuration.

If we need to use config object, we still can combine both approaches: 
```
bind {config:FluencyConfig => Fluency.newBuilder.withConfig(config).withErrorHandler(...).build }
```

I think the configuration parameters we will use frequently include:
- useExtendedEventTime
- maxBufferSize
- flushIntervalMillis
- jvmHeapBufferMode = true (I'm not sure how safe it is to use DirectByteBuffer in fluency) 
- ackResponseMode = true/false (This depends on whether we can accept missing data)
- sslEnabled = true (For data security)

We also need to determine these two parameters programmatically:
- fileBackupDir (e.g., `{spark.tempdir}/{app id}/fluency`)
- errorHandler 

So instead of having nulls in config, how about adding a simple Fluency builder which accepts the above basic configuration set + withXXX methods for backup dir + error handler?

Reply:
I see. In any case, it's difficult choice in design. What about providing pre-configured `Design` as follows?
```scala
package object fluentd {
  def withFluency(host: String = "127.0.0.1",
                  port: Int = 24224,
                  useExtendedEventTime: Boolean = false,
                  maxBufferSize: Long = 512 * 1024 * 1024,
                  flushIntervalMillis: Int = 600,
                  jvmHeapBufferMode: Boolean = true,
                  ackResponseMode: Boolean = true,
                  sslEnabled: Boolean = true): Design = ...
}
```
Users can simply override above properties as they needed. But still we need to consider how to configure `fileBackupDir` and `errorHandler` programmatically. Builder might be necessary anyway.